src/libsyntax/diagnostic_list.rs

```diff
@@ -93,6 +93,38 @@ For more information about the cfg attribute, read:
 https://doc.rust-lang.org/reference.html#conditional-compilation
 "##,
 
+E0541: r##"
+An unknown meta item was used.
+
+Erroneous code example:
+
+```compile_fail,E0541
+#[deprecated(
+    since="1.0.0",
+    // error: unknown meta item
+    reason="Example invalid meta item. Should be 'note'")
+]
+fn deprecated_function() {}
+```
+
+Meta items are the key/value pairs inside of an attribute. The keys provided
+must be one of the valid keys for the specified attribute.
+
+To fix the problem, either remove the unknown meta item, or rename it if you
+provided the wrong name.
+
+In the erroneous code example above, the wrong name was provided, so changing it
+to the right name fixes the error.
```

Comment:
Please replace "error." by "error:".

Reply:
Also, the formulation isn't very nice. What about:

"In the erroneous code example above, the wrong name was provided, so changing to a correct one it will fix the error. Example:"

?